src/main/java/lambdaperson/FilteredPersons.java

```diff
@@ -0,0 +1,23 @@
+package lambdaperson;
+
+import personlist.Person;
+import java.util.Comparator;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class FilteredPersons {
+
+    public static Set<Person> filterPersons(Set<Person> personList) {
+        Comparator<Person> sortingByNameLength =
+                Comparator.comparingInt((Person p) -> p.getName().length());
+
+        return personList.stream().filter(Person -> Person.getAge() < 30 & Person.getAge() > 20)
+                .sorted(sortingByNameLength)
+                .map((Person p) -> {
+                    p.setName(p.getName().toUpperCase());
+                    return p;
+                })
+                .collect(Collectors.toCollection(LinkedHashSet::new));
```

Comment:
So we need to return a sorted collection of smth.
Ther're two main approaches to do that:
1. Return a List that is preliminary sorted
2. Return a Set with TreeSet instantiated with required comparator

Returning a `LinkedHashSet` is technically correct, but seems a bit odd for a general case. Such approach might be used though for some specific case...